pkg/raft/raft.go

```diff
@@ -611,6 +615,28 @@ func (r *raft) send(m pb.Message) {
 	}
 }
 
+// makeMsgApp constructs a MsgApp message for being sent to the given peer, and
+// hands it over to the caller. Updates the replication flow control state to
+// account for the fact that the message is about to be sent.
+func (r *raft) makeMsgApp(to pb.PeerID, pr *tracker.Progress, ls logSlice) pb.Message {
```

Comment:
the side effects here are a little unusual and could potentially trip someone up in the future. I don't have a good idea on how to improve, though, other the somewhat lame rename `makeMsgAppMustSend`. Making a specialized send method just for `MsgApp` would be a decent alternative if we didn't use the same `pb.Message` for everything.
I'm fine with this as is but if you have ideas to improve this I'd be grateful.
Should not block this PR.

Reply:
Hmm, I guess messages can always be dropped and so while it'd be really unhelpful to call this method, it wouldn't be "incorrect". So "must send" is a little too strong. Hmm.

Reply:
We could also split it into 2 funcs: `makeMsgApp` and `sendMsgApp`. But at the moment there are no cases when we create a message and then not "send" it, so it was ok to merge the two.

Reply:
Renamed to `prepareMsgApp` which should sound less confusing.